docs/features/role-based-access-control/high-level-design.md

```diff
@@ -0,0 +1,100 @@
+# Role base access control - High-level design
+
+This document aims at providing an overview of the architecture of Role-based Access Control (RBAC from here on) for
+the Marxan platform.
+
+## Roles
+
+Platform administrators should be identified by a boolean attribute in the users table (`is_admin`). This attribute can
+only be edited by other platform administrators, and it must not be editable by the user when editing their profile.
+
+The remaining roles supported by the platform (`Owner`, `Contributor` and `Viewer`) will be stored in the `roles` 
+database in the `marxan-api` database.
+
+Roles should have numerical values associated to them, to identify the hierarchical order between them. For instance:
+
+* `Owner` -> 1000
+* `Contributor` -> 500
+* `Viewer` -> 250
+
+Roles with fewer privileges should have lower numerical values than roles with more privileges. Using numerical values
+for roles allows for easily comparing if a given user has certain privileges (e.g. checking if a user has the role 
+`Contributor` can be easily done: `role >= Contributor (500)`).
+
+There is also an implicit relationship between platform actions and the supported roles. By default, it is assumed that:
+  * a user with role `Viewer` can read the information for the given resource (which usually includes calls to 
+`GET resource` and `GET resource/:id` endpoints).
+  * user with role `Contributor` can perform all the actions a `Viewer` can, as well as edition actions for the given
+resource (which usually includes calls to `PATCH/PUT resource/:id` endpoints).
+  * user with `Owner` role can perform all the actions a `Contributor` can, as well as deleting actions for the given
+resource (which usually includes calls to `DELETE resource/:id` endpoints).
+
+**However, it is up to each controller method to correctly check permissions for the action being performed.** This also applies
+to custom actions that might be subject to RBAC: for instance, if publishing a project is a custom action that should be
+subject to RBAC, then it is up to the controller action to check that the user performing the request has the required role 
+(e.g. `Owner`) to perform that action.
+
+The mapping between the different platform actions and the role required to perform such action is kept in the following
+spreadsheet: https://docs.google.com/spreadsheets/d/1vz-YHdoAAG2nv_B1QUR7LQ-eU_DnKBexrqULrzMMOtw/edit?usp=sharing
```

Comment:
As we want to keep HLD the one and only single source of truth, let's avoid linking external documents:
* they may get obsolete
* they may be no longer accessible
* they may be not open for OS contributors
* documents are easy to change without developers knowledge, thus making implementation out of date
* its hard to track changes in the external document 
* keeping it on repository makes it easier to control the flow (everyone can see PR with changes to requirements, approve it, discuss etc.)

Reply:
Makes sense, I'll move it here 👍